src/status_im/communities/core.cljs

```diff
@@ -784,3 +778,43 @@
                                 :community-id community-id
                                 :public-key   public-key
                                 :role-id      role-id})]})
+
+(rf/defn handle-fetched-collapsed-community-categories
+  {:events [:communities/fetched-collapsed-community-categories]}
+  [{:keys [db]} categories]
+  {:db
+   (assoc
+    db
+    :communities/collapsed-categories
+    (reduce
+     (fn [acc {:keys [communityId categoryId]}]
+       (assoc-in acc [communityId categoryId] true))
+     {}
+     categories))})
+
+(rf/defn fetch-collapsed-community-categories
+  [_]
+  {:json-rpc/call [{:method     "wakuext_collapsedCommunityCategories"
+                    :params     []
+                    :on-success #(re-frame/dispatch
+                                  [:communities/fetched-collapsed-community-categories %])
+                    :on-error   #(log/error "failed to fetch collapsed community categories" %)}]})
+
+(rf/defn handle-toggled-collapse-category
+  {:events [:communities/toggled-collapse-category]}
+  [{:keys [db]} community-id category-id collapsed?]
+  {:db (assoc-in db [:communities/collapsed-categories community-id category-id] collapsed?)})
+
+(rf/defn handle-toggle-collapse-category
+  {:events [:communities/toggle-collapse-category]}
+  [{:keys [db]} community-id category-id collapse?]
+  {:json-rpc/call [{:method     "wakuext_toggleCollapsedCommunityCategory"
+                    :params     [{:communityId community-id
+                                  :categoryId  category-id
+                                  :collapsed   collapse?}]
+                    :on-success #(re-frame/dispatch
+                                  [:communities/toggled-collapse-category
+                                   community-id
+                                   category-id
+                                   collapse?])
+                    :on-error   #(log/error "failed to toggle collapse category" %)}]})
```

Comment:
Just a light suggestion.

I've been logging a bit differently, enriching a bit more the logs. Usually in the error event I pass not only the error instance, but any extra bits of data that could be relevant to investigate, such as the community ID. Of course, in your case here you'd need to get a handle on the community ID.

```clojure
(log/error "Failed to toggle collapsed category"
           {:error        %
            :community-id community-id
            :category-id category-id})
```

Reply:
Just a heads up, not something you should do, but I started to add the `:event` keyword to the map as well, to log the event that triggered the error. Many times, just from the log alone I don't know the name of the event that originated the error. Surely we can search for the error string.